File: Hackerrank/30-Days/day-29/solution.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class Solution {



    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {
        int t = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

        for (int tItr = 0; tItr < t; tItr++) {
            String[] nk = scanner.nextLine().split(" ");

            int n = Integer.parseInt(nk[0]);

            int k = Integer.parseInt(nk[1]);
            
            Process(n,k);
        }

        scanner.close();
    }
    
    public static void Process(int n, int k){
        int max = -1;
        for (int i=1;i<=n-1;i++){
            for (int j=i+1;j<=n;j++){
                int temp = i&j;
                if (temp<k){
                    if (max < temp){
                        max = temp;
                    }
                }
            }
        }
        System.out.println(max);
    }
}

File: Hackerrank/30-Days/day-29/solution.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));

    main();
});

function readLine() {
    return inputString[currentLine++];
}

function processData(n, k){
    var max = Number.MIN_SAFE_INTEGER;
    for(var i=1;i<=n-1;i++){
        for(var j=i+1;j<=n;j++){
            var temp = i&j;
            if(temp < k && max < temp)
                max = temp;
        }
    }
    console.log(max);
}

function main() {
    const t = parseInt(readLine(), 10);

    for (let tItr = 0; tItr < t; tItr++) {
        const nk = readLine().split(' ');

        const n = parseInt(nk[0], 10);

        const k = parseInt(nk[1], 10);
        
        processData(n, k);
    }
}
